resources/js/Components/ComponentsAssistants/SoalInputPilihanGanda.jsx

```diff
@@ -0,0 +1,266 @@
+import { useState } from "react";
+import ModalDelateSoal from "./ModalDelateSoal";
+import ModalEditSoalPG from "./ModalEditSoalPG";
+import ModalSaveSoal from "./ModalSaveSoal";
+import trashIcon from "../../../assets/nav/Icon-Delete.svg";
+import editIcon from "../../../assets/nav/Icon-Edit.svg";
+
+export default function SoalInputPG({ tipeSoal, modul, onModalSuccess, onModalValidation, addSoal, }) {
+    const [soal, setSoal] = useState([""]);
+    const [pilihan, setPilihan] = useState(["", "", "", ""]);
+    const [soalList, setSoalList] = useState([]);
+    const [isModalOpenDelate, setIsModalOpenDelate] = useState(false);
+    const [selectedNomor, setSelectedNomor] = useState(null);
+    const [isModalOpenSuccess, setIsModalOpenSuccess] = useState(false);
+    const [isModalOpenEdit, setIsModalOpenEdit] = useState(false);
+    const [editingSoal, setEditingSoal] = useState(null);
+    const [mode, setMode] = useState("text");
+    const [deskripsi, setDeskripsi] = useState(["", "", ""]);
+    const [soalCounter, setSoalCounter] = useState(1); // jgn di ilangin buat counter global untuk soal
+
+    const handlePilihanChange = (index, value) => {
+        const updatedPilihan = [...pilihan];
+        updatedPilihan[index] = value;
+        setPilihan(updatedPilihan);
+    };
+
+    const handleDeskripsiChange = (index, value) => {
+        const updatedDeskripsi = [...deskripsi];
+        updatedDeskripsi[index] = value;
+        setDeskripsi(updatedDeskripsi);
+    };
+
+    const handleSoalChange = (index, value) => {
+        const updatedSoal = [...soal];
+        updatedSoal[index] = value;
+        setSoal(updatedSoal);
+    };
+
+    const handleTambahSoal = () => {
+        if (mode === "text" && soal.some((item) => !item)) {
+            onModalValidation();
+            return;
+        }
+        if (mode === "kode" && deskripsi.some((item) => !item)) {
+            onModalValidation();
+            return;
+        }
+        if (pilihan.some((pil) => !pil)) {
+            onModalValidation();
+            return;
+        }
+
+        const soalBaru = {
+            nomor: soalCounter, // counter soal
+            tipeSoal,
+            modul,
+            soal: mode === "text" ? soal : deskripsi,
+            pilihan,
+            mode,
+        };
+
+        setSoalList([...soalList, soalBaru]);
+        setSoal([""]);
+        setPilihan(["", "", "", ""]);
+        setDeskripsi(["", "", ""]);
+        setSoalCounter(soalCounter + 1); // pake increment ( 0 + 1 = 1)
```

Comment:
```suggestion
        setSoalCounter(soalCounter + 1);
```
i think lets just remove the comment, its not really usefull here 😅 , thanks!

Reply:
Oh sorry, I forgot to delete that. That's actually a comment I added yesterday when I was trying out the logic 😅